### PR TITLE
Make fileBlobUrl and fileRawUrl optional; Fix #339.

### LINKED ISSUE #339: Diff errors when commits include removed submodule
The following returns an error:
```
GitHub.Commits.diff "scott-fleischman" "repo-remove-submodule" "d03c152482169d809be9b1eab71dcf64d7405f76" "42cfd732b20cd093534f246e630b309186eb485d"
```
```
Left (ParseError "Error in $[2]['blob_url']: failed to parse field files: expected URL, encountered Null")
```

You can see the example here with the response JSON:
https://github.com/scott-fleischman/repo-remove-submodule

Would you be ok with a change like the following? If so I can put together a PR.

```
data File = File
    { fileBlobUrl   :: !(Maybe URL)
    , fileStatus    :: !Text
    , fileRawUrl    :: !(Maybe URL)
    , fileAdditions :: !Int
```

### PR DESCRIPTION
**Pull request by scott-fleischman**
This is a breaking change to the `File` type.

These fields are null for the diff api when a commit removes a submodule.

Add test that uses the repository I created to test this issue.

Remove two redundant imports to fix warnings.

### CODE REVIEW

**Comment by scott-fleischman** [comment]
This resolves #339

File: src/GitHub/Data/GitData.hs [hunk 1]
```diff
@@ -184,9 +184,9 @@ instance NFData GitUser where rnf = genericRnf
 instance Binary GitUser
 
 data File = File
-    { fileBlobUrl   :: !URL
+    { fileBlobUrl   :: !(Maybe URL)
```
Comment (phadej) [inline src/GitHub/Data/GitData.hs]: won't we be on safer ground if `FromJSON` instance is changed to use `:?` for changed fields?

Otherwise LGTM

Comment (scott-fleischman) [reply src/GitHub/Data/GitData.hs]: Good catch. Added in b4422b019605d6e60e5f4b612413bd0e79fc3a59

**Comment by scott-fleischman** [comment]
@phadej Would you mind merging this or letting me know what needs changing? Currently we are using a fork of this project.

**Pull request merged**